### DISCUSSION TITLE
Including frame number in saved crop name

### DISCUSSION THREAD

**Discussion by solomonnsumba** [ultralytics/yolov5#9769] [Q&A]
How can I include the frame number in the name of the saved crop during object detection inference on a video?

**Comment by glenn-jocher** [comment]
👋 Hello! Thanks for asking about **cropping results** with YOLOv5 🚀.  Cropping bounding box detections can be useful for training classification models on box contents for example. This feature was added in PR https://github.com/ultralytics/yolov5/pull/2827. You can crop detections using either detect.py or YOLOv5 PyTorch Hub:

## detect.py
Crops will be saved under `runs/detect/exp/crops`, with a directory for each class detected.
```bash
python detect.py --save-crop
```
<img width="450" alt="Original" src="https://user-images.githubusercontent.com/26833433/139577918-65cc8f5d-741d-4a88-8753-0154013c624d.jpg">

<img width="1067" alt="Crop" src="https://user-images.githubusercontent.com/26833433/139577896-aa26b07c-466d-4057-9bfe-b01b56007587.png">

## YOLOv5 PyTorch Hub
Crops will be saved under `runs/detect/exp/crops` if `save=True`, and also returned as a dictionary with crops as numpy arrays.
```python
import torch

# Model
model = torch.hub.load('ultralytics/yolov5', 'yolov5s')  # or yolov5m, yolov5l, yolov5x, custom

# Images
img = 'https://ultralytics.com/images/zidane.jpg'  # or file, Path, PIL, OpenCV, numpy, list

# Inference
results = model(img)

# Results
crops = results.crop(save=True) 
# -- or --
crops = results.crop(save=True, save_dir='runs/detect/exp')  # specify save dir
```

Good luck 🍀 and let us know if you have any other questions!
